### PR TITLE
[ML]add InferenceAction request query validation

### PR DESCRIPTION
**Pull request by Huaixinww**
When testing Cohere's rerank function, we discovered that if a query is not provided, cohereRerankRequest will throw a NPE. 
The validate() method of InferenceAction checks the input but does not validate the query, causing cohereRerankRequest to throw a NPE because it requires the query to be non-null.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/InferenceAction.java [hunk 1]
```diff
@@ -173,6 +173,18 @@ public ActionRequestValidationException validate() {
                 e.addValidationError("input array is empty");
                 return e;
             }
+            if (taskType.equals(TaskType.RERANK)) {
+                if (query == null) {
+                    var e = new ActionRequestValidationException();
+                    e.addValidationError("missing query");
```
Comment (timgrein) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/InferenceAction.java:179]:
```suggestion
                    e.addValidationError(format("Field [query] cannot be null for task type [%s]", TaskType.RERANK));
```

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/InferenceAction.java [hunk 2]
```diff
@@ -173,6 +173,18 @@ public ActionRequestValidationException validate() {
...
+                    return e;
+                }
+                if (query.isEmpty()) {
+                    var e = new ActionRequestValidationException();
+                    e.addValidationError("query is empty");
```
Comment (timgrein) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/InferenceAction.java:184]: e.addValidationError(format("Field [query] cannot be empty for task type [%s]", TaskType.RERANK));

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/inference/action/InferenceActionRequestTests.java [hunk 3]
```diff
@@ -71,6 +72,84 @@ public void testParsing() throws IOException {
         }
     }
 
+    public void testValidation() throws IOException {
```
Comment (timgrein) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/inference/action/InferenceActionRequestTests.java:75]: I think this test could be split into several smaller tests `testValidation_TextEmbedding_Null`, `testValidation_TextEmbedding_Empty`, `testValidation_Rerank_Null` etc.

**Review by timgrein — COMMENTED** [review]
Hi @Huaixinww 

Thank you for the PR! It would be great, if you could sign the contributor agreement to move forward on the review. For changes in the inference api you can also add the two label `:ml` and `Team:ML` to request a review from the ML team.

**Comment by Huaixinww** [comment]
Hi @timgrein 
Thank you for your suggestion! I've signed the contributor agreement, formatted the error message, and split the test into several smaller tests.

**Review by davidkyle — APPROVED** [review]
LGTM

Thank you for finding the bug and fixing it

**Comment by davidkyle** [comment]
@elasticmachine test this please

**Review by jonathan-buttner — APPROVED** [review]

**Review by timgrein — APPROVED** [review]
Thank you for the changes! LGTM 🚢

**Pull request merged**